src/transformers/generation_tf_utils.py

```diff
@@ -1287,6 +1332,28 @@ def adjust_logits_during_generation(
         elif cur_len == max_length - 1 and forced_eos_token_id is not None:
             vocab_range = tf.constant(range(vocab_size))
             return tf.where(vocab_range != forced_eos_token_id, -1e8, logits)
+
+        elif suppress_tokens is not None:
```

Comment:
I actually don't think we need those statements here in the code as the `adjust_logits_during_generation` is just a hack we've implemented for Marian if I recall correctly. So I think just having them implemented as logits processors should be good enough. cc @gante